### PR TITLE
dotCMS/core#25637 Replace DotEmptyExperiment with DotEmptyContainer

### LINKED ISSUE #25637: [UI] Replace the Empty List of Experiments with `DotEmptyContainerComponent`
Replace the Empty Component
Screenshot 2023-07-26 at 2 11 31 PM

With `DotEmptyContainerComponent`
Screenshot 2023-07-26 at 2 19 21 PM

Same content.

### PR DESCRIPTION
**Pull request by oidacra**
### Proposed Changes
* Replace DotEmptyExperiment with DotEmptyContainer

### <samp>🤖 Generated by Copilot at 458c220</samp>

Refactored the dot-experiments-list and dot-experiments-list-table components to use the dot-empty-container component and the DotMessageService for displaying a consistent and localized message when there are no experiments. Removed the unused and redundant DotExperimentsEmptyExperimentsComponent and its files. Added some styles to improve the layout of the table.

## Related Issue
Fixes #25637 

# Explanation of Changes

### <samp>🤖 Generated by Copilot at 458c220</samp>

*  Replace `dot-experiments-empty-experiments` component with `dot-empty-container` component in `dot-experiments-list-table` and `dot-experiments-list` components ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-47ff8e7da16ff598a09d04b521a054c50ccbca38ae913fec184a9440695f8b3fL54-R54), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-16cc835daf57e6a1bdb094153ddceaa113f18de6ce6d05e3526e282877f747e5L41-R45), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-1f88958edd07bf54c9d3aab3eb150cb7f144525caed75a3e95708761aff5ccf9L21-L22), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-1f88958edd07bf54c9d3aab3eb150cb7f144525caed75a3e95708761aff5ccf9L79), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-1f88958edd07bf54c9d3aab3eb150cb7f144525caed75a3e95708761aff5ccf9L104-R103), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL11-R22), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL27), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL34-R40), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL43-R55), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-e5bb9795b87f29f7f1b50486583e70f6b628c0621a94f61480d7f49633c04b52L31), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-e5bb9795b87f29f7f1b50486583e70f6b628c0621a94f61480d7f49633c04b52L59-L62), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-e5bb9795b87f29f7f1b50486583e70f6b628c0621a94f61480d7f49633c04b52L107-R102), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbL22-R26), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbL41), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbL49-R49), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbR63-R67))
* Add `height: 100%` style to `dot-experiments-list-table` component to fill the available space ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-7bbe94897b90084db6de4f17bb218eac72410540e283adac4f93d03f56c8bffcL4-R9))
* Import and use `DotMessageService` in `dot-experiments-list-table` and `dot-experiments-list` components to get the empty container message from the i18n service ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL2-R9), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL11-R22), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL43-R55), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbL5-R5), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbR14), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbR63-R67))
* Import and use `inject` function in `dot-experiments-list-table` and `dot-experiments-list` components to simplify the dependency injection in the tests ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-9c55858c58e4b1c77b3598923b9c6e6dcf5dc733db9390077691b89d2a86688aL2-R9), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-d30ecafe1ec95964e31deee140285a237d94fa60061348958bbd3b0957d04bfbL5-R5))
* Change the order of the attributes in the `p-confirmDialog` element in `dot-experiments-list` component template to match the alphabetical order ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-16cc835daf57e6a1bdb094153ddceaa113f18de6ce6d05e3526e282877f747e5L51-R53))
* Update the tests for the skeleton and status filter components in `dot-experiments-list` component spec file to use the `spectator.query` function instead of the variables ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-e5bb9795b87f29f7f1b50486583e70f6b628c0621a94f61480d7f49633c04b52L99-R95), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-e5bb9795b87f29f7f1b50486583e70f6b628c0621a94f61480d7f49633c04b52L121-R113))
* Add `dot-empty-container` component style to center the content and fill the height of the host element ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-06e64900f59e52aad3b83fa285fc8ea30ad7fef875aa01496bbfb28d11b2e447R3-R9))
* Delete the files related to the `dot-experiments-empty-experiments` component, as it is no longer used ([link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-b4ec8fc236efa115118d1b3ddac54955ca97fe5a6196790138339ff048e8d04d), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-b552aadae608fde32aa357be41d4ced6b1d1863c59211c1e4e5552dc2c7fcc57), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-cb922089790d96851e8b01148b8ab38b1a3dbe8b7eb660bb2b2ee7326a3028c2), [link](https://github.com/dotCMS/core/pull/25744/files?diff=unified&w=0#diff-2492b610748b251b677469aaca9647759424ffd95ea0d0863358b2108cf24c1f))

### CODE REVIEW

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-list/dot-experiments-list.component.ts [hunk 1]
```diff
@@ -60,6 +60,11 @@ export class DotExperimentsListComponent {
     );
     statusOptionList = ExperimentsStatusList;
     confirmDialogKey = CONFIGURATION_CONFIRM_DIALOG_KEY;
+    private dotMessageService: DotMessageService = inject(DotMessageService);
```
Comment (zJaaal) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-list/dot-experiments-list.component.ts:63]: Do we need this in this case? You have the constructor method declared

**Review by zJaaal — APPROVED** [review]
See my suggestions

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-list/dot-experiments-list.component.ts (hunk 1 above)
Comment (oidacra) [reply core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-list/dot-experiments-list.component.ts]: Removed.

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-list/components/dot-experiments-list-table/dot-experiments-list-table.component.scss [hunk 2]
```diff
@@ -1,10 +1,12 @@
 @use "variables" as *;
 
 :host {
+    height: 100%;
     .dot-experiment-list-table {
```
Comment (fmontes) [inline core-web/libs/portlets/dot-experiments/portlet/src/lib/dot-experiments-list/components/dot-experiments-list-table/dot-experiments-list-table.component.scss:5]:
```suggestion

    .dot-experiment-list-table {
```

**Review by fmontes — APPROVED** [review]

**Pull request merged**